claims.go

```diff
@@ -130,6 +134,27 @@ func (c *RegisteredClaims) VerifyIssuer(cmp string, req bool) bool {
 	return verifyIss(c.Issuer, cmp, req)
 }
 
+func (c *RegisteredClaims) validateAudience(req bool, opts ...validationOption) bool {
```

Comment:
Is it somehow possible to move this function to our `validation` struct? Otherwise we are duplicating a lot of code. This would also make sense to move the other validate/verify functions there in the long run.

Reply:
Unfortunately, it cannot be moved as each `VerrifyAudience` call is a receiver method on the respective struct. I refactored most of the code into a validator function to reduce duplication.